### PR TITLE
Extend validator support to local links

### LINKED ISSUE #87: Add config option to skip local link checks
We currently can only ignore remote links based on regex, but there are use cases where users might also want to ignore local links.

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR extends validator support for local links.

Fixes #87.

### CODE REVIEW

File: pkg/mdformatter/linktransformer/config.go [hunk 1]
```diff
@@ -52,12 +55,22 @@ type IgnoreValidator struct {
 	_regex *regexp.Regexp
 }
 
+type LocalValidator struct {
+	_regex *regexp.Regexp
+}
+
+type IgnoreLocalValidator struct {
+	_regex *regexp.Regexp
```
Comment (bwplotka) [inline pkg/mdformatter/linktransformer/config.go:63]: Let's use exactly the same validator routines for local links, so we don't need to add any extra config bit.

Probably to ensure compatibility with current use cases, we might want to add special flag for it `--enable-explicit-local-validators`

**Review by bwplotka — COMMENTED** [review]
Thanks! Some idea first to check (:

File: pkg/mdformatter/linktransformer/config.go (hunk 1 above)
Comment (saswatamcode) [reply pkg/mdformatter/linktransformer/config.go]: Keeping it as a YAML config option instead, as we mention validators via that too. 🙂

File: pkg/mdformatter/linktransformer/validator.go [hunk 2]
```diff
@@ -32,8 +32,31 @@ func (v GitHubPullsIssuesValidator) IsValid(k futureKey, r *validator) (bool, er
 	return false, nil
 }
 
-// RoundTripValidator.IsValid returns true if url is checked by colly.
+// RoundTripValidator.IsValid returns true if url is checked by colly or it is a valid local link.
 func (v RoundTripValidator) IsValid(k futureKey, r *validator) (bool, error) {
+	matches := remoteLinkPrefixRe.FindAllStringIndex(k.dest, 1)
+	if matches == nil && r.validateConfig.ExplicitLocalValidators {
```
Comment (bwplotka) [inline pkg/mdformatter/linktransformer/validator.go:38]: Does it make sense to pack it in separate function? Looks like 1:1 to repetition.

**Review by bwplotka — APPROVED** [review]
LGTM, just one nit, not a blocker - feel free to merge.

**Review by bwplotka — APPROVED** [review]

**Pull request merged**